### ISSUE TITLE
`Net::WriteTimeout with #<TCPSocket:(closed)> (Faraday::TimeoutError)` error when trying to upload an APK

### ISSUE THREAD

**Issue by spatenotte** [fastlane/fastlane-plugin-firebase_app_distribution#279]
### Issue Description
When trying to upload an APK to App Distribution, the upload starts correctly, but always fails before completion with the following error:
<img width="1585" alt="image" src="https://user-images.githubusercontent.com/5433694/185264944-10c6ee0e-7de6-486e-9d92-75db65e37d42.png">

##### Configuration
```
# Upload to Firebase
      firebase_app_distribution(
        android_artifact_type: 'APK',
        android_artifact_path: apk_path,
        app: app.firebase_android_app_id,
        firebase_cli_token: @env_manager.firebase_token,
        groups: 'everyone',
        release_notes: changelog
      )
```

##### Complete verbose output
Displayed in the "Issue Description" screenshot

##### Environment
Output of `bundle exec fastlane env`:
<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 12.5                                        |
| Ruby                        | 2.6.10                                      |
| Bundler?                    | true                                        |
| Git                         | git version 2.35.1                          |
| Installation Source         | ~/.gem/ruby/2.6.10/bin/fastlane             |
| Host                        | macOS 12.5 (21G72)                          |
| Ruby Lib Dir                | ~/.rubies/ruby-2.6.10/lib                   |
| OpenSSL Version             | OpenSSL 1.1.1q  5 Jul 2022                  |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 13.1                                        |
| Swift Version               | 5.5.1                                       |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
[...]
      # Upload to Firebase
      firebase_app_distribution(
        android_artifact_type: 'APK',
        android_artifact_path: apk_path,
        app: app.firebase_android_app_id,
        firebase_cli_token: @env_manager.firebase_token,
        groups: 'everyone',
        release_notes: changelog
      )
[...]
```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.209.0 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

| Plugin                                    | Version | Update-Status |
| ----------------------------------------- | ------- | ------------- |
| fastlane-plugin-firebase_app_distribution | 0.3.5   | ✅ Up-To-Date  |

<details><summary><b>Loaded gems</b></summary>

| Gem                                       | Version      |
| ----------------------------------------- | ------------ |
| did_you_mean                              | 1.3.0        |
| bundler                                   | 2.2.32       |
| rake                                      | 13.0.6       |
| rexml                                     | 3.2.5        |
| CFPropertyList                            | 3.0.5        |
| concurrent-ruby                           | 1.1.10       |
| i18n                                      | 1.12.0       |
| minitest                                  | 5.16.3       |
| tzinfo                                    | 2.0.5        |
| zeitwerk                                  | 2.6.0        |
| activesupport                             | 6.1.6.1      |
| public_suffix                             | 4.0.7        |
| addressable                               | 2.8.0        |
| artifactory                               | 3.0.15       |
| atomos                                    | 0.1.3        |
| aws-eventstream                           | 1.2.0        |
| aws-partitions                            | 1.619.0      |
| aws-sigv4                                 | 1.5.1        |
| jmespath                                  | 1.6.1        |
| aws-sdk-core                              | 3.132.0      |
| aws-sdk-kms                               | 1.58.0       |
| aws-sdk-s3                                | 1.114.0      |
| babosa                                    | 1.0.4        |
| circleci                                  | 2.1.0        |
| claide                                    | 1.1.0        |
| colored                                   | 1.2          |
| colored2                                  | 3.1.2        |
| highline                                  | 2.0.3        |
| commander                                 | 4.6.0        |
| date                                      | 3.2.2        |
| declarative                               | 0.0.20       |
| digest-crc                                | 0.6.4        |
| unf_ext                                   | 0.0.8.2      |
| unf                                       | 0.1.4        |
| domain_name                               | 0.5.20190701 |
| dotenv                                    | 2.8.1        |
| emoji_regex                               | 3.2.3        |
| excon                                     | 0.92.4       |
| faraday-em_http                           | 1.0.0        |
| faraday-em_synchrony                      | 1.0.0        |
| faraday-excon                             | 1.1.0        |
| faraday-httpclient                        | 1.0.1        |
| multipart-post                            | 2.0.0        |
| faraday-multipart                         | 1.0.4        |
| faraday-net_http                          | 1.0.1        |
| faraday-net_http_persistent               | 1.2.0        |
| faraday-patron                            | 1.0.0        |
| faraday-rack                              | 1.0.0        |
| faraday-retry                             | 1.0.3        |
| ruby2_keywords                            | 0.0.5        |
| faraday                                   | 1.10.1       |
| http-cookie                               | 1.0.5        |
| faraday-cookie_jar                        | 0.0.7        |
| faraday_middleware                        | 1.2.0        |
| fastimage                                 | 2.2.6        |
| gh_inspector                              | 1.1.3        |
| jwt                                       | 2.4.1        |
| memoist                                   | 0.16.2       |
| multi_json                                | 1.15.0       |
| os                                        | 1.1.4        |
| signet                                    | 0.17.0       |
| googleauth                                | 1.2.0        |
| httpclient                                | 2.8.3        |
| mini_mime                                 | 1.1.2        |
| trailblazer-option                        | 0.1.2        |
| uber                                      | 0.1.0        |
| representable                             | 3.2.0        |
| retriable                                 | 3.1.2        |
| webrick                                   | 1.7.0        |
| google-apis-core                          | 0.7.0        |
| google-apis-androidpublisher_v3           | 0.25.0       |
| google-apis-playcustomapp_v1              | 0.10.0       |
| google-apis-iamcredentials_v1             | 0.13.0       |
| google-apis-storage_v1                    | 0.17.0       |
| google-cloud-env                          | 1.6.0        |
| google-cloud-errors                       | 1.2.0        |
| google-cloud-core                         | 1.6.0        |
| google-cloud-storage                      | 1.38.0       |
| json                                      | 2.6.2        |
| mini_magick                               | 4.11.0       |
| naturally                                 | 2.2.1        |
| optparse                                  | 0.1.1        |
| plist                                     | 3.6.0        |
| rubyzip                                   | 2.3.2        |
| security                                  | 0.1.3        |
| simctl                                    | 1.6.8        |
| terminal-notifier                         | 2.0.0        |
| unicode-display_width                     | 1.8.0        |
| terminal-table                            | 1.8.0        |
| tty-screen                                | 0.8.1        |
| tty-cursor                                | 0.7.1        |
| tty-spinner                               | 0.9.3        |
| word_wrap                                 | 1.0.0        |
| nanaimo                                   | 0.3.0        |
| xcodeproj                                 | 1.22.0       |
| rouge                                     | 2.0.7        |
| xcpretty                                  | 0.3.0        |
| xcpretty-travis-formatter                 | 1.0.1        |
| fastlane-plugin-firebase_app_distribution | 0.3.5        |
| mime-types-data                           | 3.2022.0105  |
| mime-types                                | 3.4.1        |
| multi_xml                                 | 0.6.0        |
| httparty                                  | 0.20.0       |
| mini_portile2                             | 2.8.0        |
| racc                                      | 1.6.0        |
| nokogiri                                  | 1.13.8       |
| stringio                                  | 3.0.2        |
| time                                      | 0.2.0        |
| uri                                       | 0.11.0       |
| open-uri                                  | 0.2.0        |
| open3                                     | 0.1.1        |
| ruby-progressbar                          | 1.11.0       |
</details>

*generated on:* **2022-08-17**
</details>

**Comment by jladieu** [comment]
Hi @spatenotte thanks for reaching out.  I'm an engineer on the team that supports Firebase App Distribution and I'd be happy to help you out with this.  

Is this issue still happening persistently for you?

Would you be willing to come join us on the [Firebase Community Slack](https://firebase.community/) and DM me or post to our channel #app-distribution to exchange more detailed information?  Specifically I'm thinking I could take a look at your APK to see if there are any particular qualities that might cause this unexpected timeout.

**Comment by emthomas** [comment]
+1

**Comment by jladieu** [comment]
@emthomas If this is happening to you too, I'd similarly invite you to our Firebase Community Slack and the #app-distribution channel where I or one of our team members can connect directly to try some troubleshooting steps.

**Comment by spatenotte** [comment]
Thanks @jladieu, adding the link to my Slack message here for visibility: https://firebase-community.slack.com/archives/CJU4CK4NT/p1661791958830069

**Comment by kaibolay** [comment]
This should be fixed with #281

**Comment by spatenotte** [comment]
Thank you team, I've verified that `v0.3.6` fixes my issue! 🙌
